src/binary/ContractionBackend.h

```diff
@@ -0,0 +1,271 @@
+    uint64_t m_lda = 0;
+    //! kernel leading dimension B
+    uint64_t m_ldb = 0;
+    //! kernel leading dimension C
+    uint64_t m_ldc = 0;
+
+    //! kernel m stride of auxiliary tensor
+    uint64_t m_stride_m_out_aux = 0;
+    //! kernel n stride of auxiliary tensor
+    uint64_t m_stride_n_out_aux = 0;
+
+    //! kernel br stride a
+    uint64_t m_br_stride_a = 0;
+    //! kernel br stride b
+    uint64_t m_br_stride_b = 0;
+
+    //! complex stride of the left tensor
+    int64_t m_cpx_stride_in_left_bytes = 0;
+    //! complex stride of the right tensor
+    int64_t m_cpx_stride_in_right_bytes = 0;
+    //! complex stride of the auxiliary output tensor
+    int64_t m_cpx_stride_out_aux_bytes = 0;
+    //! complex stride of the output tensor
+    int64_t m_cpx_stride_out_bytes = 0;
+
+    //! indicates if kernel should transpose A
+    bool m_trans_a = false;
+    //! indicates if kernel should transpose B
+    bool m_trans_b = false;
+    
+  public:
+    /**
+     * Initializes the class.
+     *
+     * @param i_dim_type dimension types.
+     * @param i_exec_type execution types.
+     * @param i_dim_sizes sizes of the dimensions.
+     * @param i_strides_left strides in the left input tensor.
+     * @param i_strides_right strides in the right input tensor.
+     * @param i_strides_out_aux strides in the auxiliary output tensor.
+     * @param i_strides_out strides in the output tensor.
+     * @param i_dtype_left datatype of left input tensor.
+     * @param i_dtype_right datatype of right input tensor.
+     * @param i_dtype_comp datatype of computation.
+     * @param i_dtype_out datatype of output tensor.
+     * @param i_ktype_first_touch type of the first touch kernel.
+     * @param i_ktype_main type of the main kernel.
+     * @param i_ktype_last_touch type of the last touch kernel.
+     * @param i_num_threads number of threads used for contraction.
+     **/
+    void init( std::vector< dim_t >   const & i_dim_type,
+               std::vector< exec_t >  const & i_exec_type,
+               std::vector< int64_t > const & i_dim_sizes,
+               std::vector< int64_t > const & i_strides_left,
+               std::vector< int64_t > const & i_strides_right,
+               std::vector< int64_t > const & i_strides_out_aux,
+               std::vector< int64_t > const & i_strides_out,
+               data_t                         i_dtype_left,
+               data_t                         i_dtype_right,
+               data_t                         i_dtype_comp,
+               data_t                         i_dtype_out,
+               kernel_t                       i_ktype_first_touch,
+               kernel_t                       i_ktype_main,
+               kernel_t                       i_ktype_last_touch,
+               int64_t                        i_num_threads );
+
+
+    /**
+     * Initializes the class with a vector of iter_propertys.
+     *
+     * @param i_loops vector of iter_propertys.
```

Comment:
Fix: The parameter name ``i_loops`` is misleading as not every dim specified has to be instantiated as a loop.